River tries to import the inexistent module compat.pytorch

<!--
Before submitting a bug, please make sure the issue hasn't been already
addressed by searching through the past issues.
-->
## Versions
<!--
Describe all the versions that can help us to correct the bug
-->

**River version**: 0.21.1
**Python version:** 3.12.5
**Operating system**: macOS 14.6.1

## Describe the bug
<!--
A clear and concise description of the bug.
-->

In `test_estimators.py`, River tries to import `PyTorch2RiverBase` from `river.compat.pytorch`.
https://github.com/online-ml/river/blob/d606d7b4b70e1867e601d77c645880eadb3ae472/river/test_estimators.py#L28-L33

The thing is, there is no file `river/compat/pytorch.py`:
```shell
$ ls river/compat
__init__.py		river_to_sklearn.py	test_sklearn.py
__pycache__		sklearn_to_river.py
```

The file was removed in commit ac5c9a1.

This raises an error in MyPy when missing module checking is enabled (such a check is disabled for River in [`pyproject.toml`](https://github.com/online-ml/river/blob/d606d7b4b70e1867e601d77c645880eadb3ae472/pyproject.toml#L131-L146)):

```shell
$ mypy river
[...]
river/test_estimators.py:29: error: Cannot find implementation or library stub for module named "river.compat.pytorch"  [import-not-found]
```

Because the module does not exist any more, the import try-except can probably be removed.

<!-- Thanks for contributing! -->

--- a/river/conftest.py
+++ b/river/conftest.py
@@ -5,7 +5,6 @@ collect_ignore = []
 try:
     import sklearn  # noqa: F401
 except ImportError:
-    collect_ignore.append("compat/sklearn.py")
     collect_ignore.append("compat/test_sklearn.py")
 
 try:
@@ -13,11 +12,6 @@ try:
 except ImportError:
     collect_ignore.append("stream/iter_sql.py")
     collect_ignore.append("stream/test_sql.py")
-
-try:
-    import torch  # noqa: F401
-except ImportError:
-    collect_ignore.append("compat/pytorch.py")
 
 try:
     import vaex  # noqa: F401

--- a/river/test_estimators.py
+++ b/river/test_estimators.py
@@ -5,6 +5,7 @@ import importlib
 import inspect
 
 import pytest
+from sklearn import linear_model as sk_linear_model
 
 from river import (
     anomaly,
@@ -24,9 +25,6 @@ from river import (
     preprocessing,
     time_series,
 )
-
-from sklearn import linear_model as sk_linear_model
-
 from river.compat.river_to_sklearn import River2SKLBase
 from river.compat.sklearn_to_river import SKL2RiverBase
 

--- a/river/test_estimators.py
+++ b/river/test_estimators.py
@@ -25,12 +25,6 @@ from river import (
     time_series,
 )
 
-try:
-    from river.compat.pytorch import PyTorch2RiverBase
-
-    PYTORCH_INSTALLED = True
-except ImportError:
-    PYTORCH_INSTALLED = False
 from sklearn import linear_model as sk_linear_model
 
 from river.compat.river_to_sklearn import River2SKLBase
@@ -83,9 +77,6 @@ def iter_estimators_which_can_be_tested():
         preprocessing.StatImputer,
         time_series.base.Forecaster,
     )
-
-    if PYTORCH_INSTALLED:
-        ignored = (*ignored, PyTorch2RiverBase)
 
     def can_be_tested(estimator):
         return not inspect.isabstract(estimator) and not issubclass(estimator, ignored)

